File: atcoder/abc131B_bite_eating.nim
# unihernandez22
# https://atcoder.jp/contests/abc131/tasks/abc131_b
# implementation

import strutils, sequtils;

var n, l: int;
(n, l) = stdin.readLine.split.map parseInt;

var eat = int.high;
var cumulative = 0;
var current: int;
for i in 1..n:
  current = l+i-1
  if abs(current) < abs(eat):
    eat = current
  cumulative += current

echo cumulative-eat

File: atcoder/abc131B_bite_eating_.ml
(* uninhm
 * https://atcoder.jp/contests/abc131/tasks/abc131_b
 * implementation
 * *)

let rec solution n l i cum eat =
    if i > n then cum-eat
    else
        let curr = i+l-1 in
        if abs curr < abs eat then
            solution n l (i+1) (cum+curr) curr
        else
            solution n l (i+1) (cum+curr) eat;;

Printf.printf "%d\n" @@
Scanf.scanf "%d %d\n" @@ fun n l ->
    solution n l 1 0 300;;

